vpr/src/device/rr_graph_view.h

```diff
@@ -96,6 +96,72 @@ class RRGraphView {
         return node_storage_.fan_in(node);
     }
 
+    /* Get the xlow of a routing resource node. This function is inlined for runtime optimization. */
+    inline short node_xlow(RRNodeId node) const {
+        return node_storage_.node_xlow(node);
+    }
+
+    /* Get the xhigh of a routing resource node. This function is inlined for runtime optimization. */
+    inline short node_xhigh(RRNodeId node) const {
+        return node_storage_.node_xhigh(node);
+    }
+
+    /* Get the ylow of a routing resource node. This function is inlined for runtime optimization. */
+    inline short node_ylow(RRNodeId node) const {
+        return node_storage_.node_ylow(node);
+    }
+
+    /* Get the yhigh of a routing resource node. This function is inlined for runtime optimization. */
+    inline short node_yhigh(RRNodeId node) const {
+        return node_storage_.node_yhigh(node);
+    }
+
+    /* Get the length of a routing resource node. This function is inlined for runtime optimization. */
+    /* node_length() only applies to CHANX or CHANY and is always a positive number*/
+    inline int node_length(RRNodeId node) const {
+        VTR_ASSERT(node_type(node) == CHANX || node_type(node) == CHANY);
+        int length = 1 + node_xhigh(node) - node_xlow(node) + node_yhigh(node) - node_ylow(node);
+        VTR_ASSERT_SAFE(length > 0);
+        return length;
+    }
+
+    /* Check if routing resource node is initialized. This function is inlined for runtime optimization. */
+    inline bool node_is_initialized(RRNodeId node) const {
+        return !((node_type(node) == NUM_RR_TYPES)
+                 && (node_xlow(node) == -1) && (node_ylow(node) == -1)
+                 && (node_xhigh(node) == -1) && (node_yhigh(node) == -1));
+    }
+
+    /* Check if two routing resource nodes are adjacent (must be a CHANX and a CHANY). This function is inlined for runtime optimization. */
+    inline bool nodes_are_adjacent(RRNodeId chanx_node, RRNodeId chany_node) const {
+        VTR_ASSERT(node_type(chanx_node) == CHANX && node_type(chany_node) == CHANY);
+        if (node_ylow(chany_node) > node_ylow(chanx_node) + 1 || // verifies that chany_node is not more than one unit above chanx_node
+            node_yhigh(chany_node) < node_ylow(chanx_node))      // verifies that chany_node is not more than one unit beneath chanx_node
+            return false;
+        if (node_xlow(chanx_node) > node_xlow(chany_node) + 1 || // verifies that chany_node is not more than one unit to the left of chanx_node
+            node_xhigh(chanx_node) < node_xlow(chany_node))      // verifies that chany_node is not more than one unit to the right of chanx_node
+            return false;
+        return true;
+    }
+
+    /* Check if node is within bounding box. This function is inlined for runtime optimization. */
```

Comment:
Expand the comment. To return true, the RRNode must be completely contained within the specified bounding box, with the edges of the bounding box being inclusive.